### PR TITLE
Text input tests

### PR DESCRIPTION
**Pull request by juanfabrega**
Added basic tests to our TextInput and FormLabel components to test:

1. Prop validation.
2. Core functionality
3. Snapshot variations

### CODE REVIEW

**Comment by juanfabrega** [comment]
https://trello.com/c/7z6sNNbp/42-textinput-component

**Comment by juanfabrega** [comment]
[TextInput Component](url)

**Comment by juanfabrega** [comment]
[TextInput Component](https://trello.com/c/7z6sNNbp/42-textinput-component)

File: src/components/TextInput/TextInput.test.jsx [hunk 1]
```diff
@@ -0,0 +1,143 @@
...
+      .toContain('Failed prop type: The prop `onChange`');
+  });
+
+  test('Throws an error if "type" prop is anything other than allowed values', () => {
+    console.error = jest.fn(); // eslint-disable-line no-console
+    render(<TextInput onChange={() => null} value="hello" type="notOnTheList" />);
+    expect(console.error).toHaveBeenCalledTimes(1); // eslint-disable-line no-console
+    expect(console.error.mock.calls[0][0]) // eslint-disable-line no-console
+      .toContain('Failed prop type: Invalid prop `type`');
+  });
```
Comment (ericyliu) [inline src/components/TextInput/TextInput.test.jsx:37]: we actually dont need these validation texts when we change this to typescript as the code wont even compile if the necessary props aren't passed in

File: src/components/TextInput/TextInput.test.jsx [hunk 2]
```diff
@@ -0,0 +1,143 @@
...
+        id="firstName"
+        label="first name"
+        value="hello"
+        onChange={onChangeMock}
+      />,
+    );
+    const inputElement = screen.getByDisplayValue('hello');
+
+    fireEvent.change(inputElement, { target: { value: 'good bye' } });
+    expect(onChangeMock).toHaveBeenCalledTimes(1);
```
Comment (ericyliu) [inline src/components/TextInput/TextInput.test.jsx:57]: should you check if it was called with the value?

File: src/components/TextInput/TextInput.test.jsx (hunk 1 above)
Comment (juanfabrega) [reply src/components/TextInput/TextInput.test.jsx:37]: Hey @ericyliu perhaps I'm misunderstanding, but the goal of these is to guard against somebody flipping a prop requirement. Basically this test breaks to let you know that the prop was required for a reason so best not to remove that requirement without explicit intent. Let me know if that makes sense.

File: src/components/TextInput/TextInput.test.jsx (hunk 2 above)
Comment (juanfabrega) [reply src/components/TextInput/TextInput.test.jsx]: @ericyliu Interestingly this seems to be the recommended approach with RTL. I actually tried checking the value but my synthetic events all come back with target of null. Not sure if this is a quirk of RTL or what but we can definitely keep investigating.

File: src/components/TextInput/TextInput.test.jsx (hunk 1 above)
Comment (ericyliu) [reply src/components/TextInput/TextInput.test.jsx:37]: ohh good point, yeah that's reasonable

File: src/components/FormLabel/FormLabel.test.jsx [hunk 3]
```diff
@@ -0,0 +1,43 @@
...
+
+  test('Matches the snapshot for component with isRequired prop', () => {
+    const { asFragment } = render(<FormLabel inputId="myId" labelText="hello world" isRequired />);
+    expect(asFragment()).toMatchSnapshot();
+  });
+
+  test('Matches the snapshot for component with hasError prop', () => {
+    const { asFragment } = render(<FormLabel inputId="myId" labelText="hello world" hasError />);
+    expect(asFragment()).toMatchSnapshot();
+  });
```
Comment (nathanyoung) [inline src/components/FormLabel/FormLabel.test.jsx:42]: remove

File: src/components/FormLabel/__snapshots__/FormLabel.test.jsx.snap [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+exports[`FormLabel Matches the snapshot for component with isRequired prop 1`] = `
+<DocumentFragment>
+  <label
+    class="label"
+    for="myId"
+  >
+    hello world
+  </label>
+</DocumentFragment>
+`;
```
Comment (nathanyoung) [inline src/components/FormLabel/__snapshots__/FormLabel.test.jsx.snap:34]: remove

File: src/components/Heading/Heading.test.jsx [hunk 5]
```diff
@@ -52,28 +52,28 @@ describe('Heading', () => {
...
   });
 
   headingComponentColors.forEach((headingComponent, index) => {
     test(`Matches the snapshot for color ${PALMETTO_BRAND_COLOR_OPTIONS[index]}`, () => {
       const { asFragment } = render(headingComponent);
-      expect(asFragment(headingComponent)).toMatchSnapshot();
+      expect(asFragment()).toMatchSnapshot();
     });
   });
 });
```
Comment (nathanyoung) [inline src/components/Heading/Heading.test.jsx:79]: unit test instead. if you want to do that in a separate pr, that works too.

Comment (juanfabrega) [reply src/components/Heading/Heading.test.jsx:79]: @nathanyoung I will do this on a separate PR.

File: src/components/FormLabel/__snapshots__/FormLabel.test.jsx.snap (hunk 4 above)
Comment (juanfabrega) [reply src/components/FormLabel/__snapshots__/FormLabel.test.jsx.snap]: @nathanyoung I have replaced these with unit tests.
